exercises/bowling/tests/bowling.rs

```diff
@@ -0,0 +1,262 @@
+#[test]
+fn twenty_zero_pin_rolls_is_zero() {
+    let game = BowlingGame::new();
+
+    for x in (0..20) {
+        game.roll(0);
+    }
+
+    assert_eq!(game.score().unwrap(), 0);
+}
+
+#[test]
+fn twenty_one_pin_rolls_is_twenty() {
+    let game = BowlingGame::new();
+
+    for x in (0..20) {
+        game.roll(1);
+    }
+
+    assert_eq!(game.score().unwrap(), 20);
+}
+
+#[test]
+fn ten_frames_without_a_strike_or_spare() {
+    let game = BowlingGame::new();
+
+    for x in (0..20) {
+        game.roll(4);
+    }
+
+    assert_eq!(game.score().unwrap(), 80);
+}
+
+#[test]
+fn spare_in_the_first_frame_followed_by_zeros() {
+    let game = BowlingGame::new();
+
+    game.roll(4);
+    game.roll(6);
+
+    for x in (0..18) {
+        game.roll(0);
+    }
+
+    assert_eq!(game.score().unwrap(), 10);
+}
+
+#[test]
+fn points_scored_in_the_roll_after_a_spare_are_counted_twice_as_a_bonus() {
+    let game = BowlingGame::new();
+
+    game.roll(5);
+    game.roll(5);
+    game.roll(3);
+
+    for x in (0..17) {
+        game.roll(0);
+    }
+
+    assert_eq!(game.score().unwrap(), 16);
+}
+
+#[test]
+fn consecutive_spares_each_get_their_bonus_points() {
+    let game = BowlingGame::new();
+
+    game.roll(5);
+    game.roll(5);
+    game.roll(3);
+    game.roll(7);
+    game.roll(4);
+
+    for x in (0..15) {
+        game.roll(0);
+    }
+
+    assert_eq!(game.score().unwrap(), 31);
+}
+
+#[test]
+fn if_the_last_frame_is_a_spare_you_get_one_extra_roll_that_is_scored_once() {
+    let game = BowlingGame::new();
+
+    for x in (0..17) {
+        game.roll(0);
+    }
+
+    game.roll(5);
+    game.roll(5);
+    game.roll(7);
+
+    assert_eq!(game.score().unwrap(), 17);
+}
+
+#[test]
+fn a_strike_earns_ten_points_in_a_frame_with_a_single_roll() {
+    let game = BowlingGame::new();
+
+    game.roll(10);
+
+    for x in (0..17) {
+        game.roll(0);
+    }
+
+    assert_eq!(game.score().unwrap(), 0);
+}
+
+#[test]
+fn points_scored_in_the_two_rolls_after_a_strike_are_counted_twice_as_a_bonus() {
+    let game = BowlingGame::new();
+
+    game.roll(10);
+    game.roll(5);
+    game.roll(3);
+
+    for x in (0..15) {
+        game.roll(0);
+    }
+
+    assert_eq!(game.score().unwrap(), 26);
+}
+
+#[test]
+fn points_scored_in_the_two_rolls_after_a_strike_are_counted_twice_with_consecutive_strikes() {
+    let game = BowlingGame::new();
+
+    game.roll(10);
+    game.roll(10);
+    game.roll(10);
+    game.roll(5);
+    game.roll(3);
+
+    for x in (0..11) {
```

Comment:
there are 6 frames left at this point, so 11 rolls isn't enough for them (0..11 is exclusive of 11 so this rolls 11 balls not 12)